File: frontend/src/types/knowledgeMap.ts
export type KnowledgeMapNodeType = 'knowledge_base' | 'knowledge_entry' | 'document' | 'tag'
export type KnowledgeMapEdgeType = 'contains' | 'cites' | 'tagged' | 'related'

export interface KnowledgeMapNode {
  id: string
  type: KnowledgeMapNodeType
  entity_id: string | null
  label: string
  metadata: Record<string, unknown>
}

export interface KnowledgeMapEdge {
  id: string
  type: KnowledgeMapEdgeType
  source: string
  target: string
  metadata: Record<string, unknown>
}

export interface KnowledgeMapResponse {
  nodes: KnowledgeMapNode[]
  edges: KnowledgeMapEdge[]
}

File: backend/app/schemas/knowledge_map.py
from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, Field

KnowledgeMapNodeType = Literal["knowledge_base", "knowledge_entry", "document", "tag"]
KnowledgeMapEdgeType = Literal["contains", "cites", "tagged", "related"]


class KnowledgeMapNode(BaseModel):
    id: str
    type: KnowledgeMapNodeType
    entity_id: UUID | None
    label: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class KnowledgeMapEdge(BaseModel):
    id: str
    type: KnowledgeMapEdgeType
    source: str
    target: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class KnowledgeMapResponse(BaseModel):
    nodes: list[KnowledgeMapNode]
    edges: list[KnowledgeMapEdge]
